signbank/dictionary/adminviews.py

```diff
@@ -329,6 +331,102 @@ def ready_for_validation_render_to_csv_response(self, context):
 
         return response
 
+    def validation_results_render_to_csv_response(self, context):
+        if not self.request.user.has_perm("dictionary.export_csv"):
+            msg = _("You do not have permissions to export to CSV.")
+            messages.error(self.request, msg)
+            raise PermissionDenied(msg)
+
+        # Create the HttpResponse object with the appropriate CSV header.
+        # It is a Python file-like object.
+        response = HttpResponse(content_type="text/csv; charset=utf-8")
+        response[
+            "Content-Disposition"
+        ] = 'attachment; filename="validation-results-export.csv"'
+
+        # Set up for outputting CSV.
+        writer = csv.writer(response)
+
+        # The queryset may or may not already be filtered for the validation:check-results tag.
+        # We have to make sure it is filtered by the tag, so we are filtering again
+        check_results_qs = TaggedItem.objects.get_by_model(
+            self.get_queryset(),
+            settings.TAG_VALIDATION_CHECK_RESULTS
+        )
+
+        sign_seen_yes_validation_records = ValidationRecord.objects.filter(
+            sign_seen=ValidationRecord.SignSeenChoices.YES
+        )
+        sign_seen_no_validation_records = ValidationRecord.objects.filter(
+            sign_seen=ValidationRecord.SignSeenChoices.NO
+        )
+        sign_seen_not_sure_validation_records = ValidationRecord.objects.filter(
+            sign_seen=ValidationRecord.SignSeenChoices.NOT_SURE
+        )
+
+        csv_queryset = (
+            check_results_qs
+            .prefetch_related(
+                Prefetch(
+                    "validation_records", queryset=sign_seen_yes_validation_records,
+                    to_attr="yes_vrs"
+                ),
+                Prefetch(
+                    "validation_records", queryset=sign_seen_no_validation_records,
+                    to_attr="no_vrs"
+                ), Prefetch(
+                    "validation_records", queryset=sign_seen_not_sure_validation_records,
+                    to_attr="not_sure_vrs"
+                ),
```

Comment:
**suggestion**: Because you're only using the count for these, it may be a good place to use [conditional aggregation](https://docs.djangoproject.com/en/5.0/ref/models/conditional-expressions/#conditional-aggregation) instead

Reply:
I am also adding comments with the respondent's name, so full row is actually needed 😅